dali/test/python/test_pipeline_debug_resnet50.py

```diff
@@ -0,0 +1,156 @@
+from time import time
+
+import nvidia.dali.fn as fn
+import nvidia.dali.types as types
+from nvidia.dali.pipeline.experimental import pipeline_def
+from test_utils import get_dali_extra_path
+
+
+@pipeline_def(device_id=0)
+def rn50_pipeline(data_path):
+    uniform = fn.random.uniform(range=(0., 1.), shape=2)
+    resize_uniform = fn.random.uniform(range=(256., 480.))
+    mirror = fn.random.coin_flip(probability=0.5)
+    jpegs, _ = fn.readers.file(
+        file_root=data_path, shard_id=0, num_shards=2)
+    images = fn.decoders.image(jpegs, output_type=types.RGB)
+    resized_images = fn.fast_resize_crop_mirror(images,
+                                                crop=(224, 224),
+                                                crop_pos_x=uniform[0],
+                                                crop_pos_y=uniform[1],
+                                                mirror=mirror,
+                                                resize_shorter=resize_uniform)
+    output = fn.crop_mirror_normalize(resized_images.gpu(),
+                                      device='gpu',
+                                      dtype=types.FLOAT16,
+                                      mean=[128., 128., 128.],
+                                      std=[1., 1., 1.])
+    return output
+
+
+@pipeline_def(device_id=0)
+def rn50_pipeline_2(data_path):
+    uniform = fn.random.uniform(range=(0., 1.), shape=2)
+    resize_uniform = fn.random.uniform(range=(256., 480.))
+    mirror = fn.random.coin_flip(probability=0.5)
+    jpegs, _ = fn.readers.file(
+        file_root=data_path, shard_id=0, num_shards=2)
+    images = fn.decoders.image(jpegs, device='mixed', output_type=types.RGB)
+    resized_images = fn.resize(images,
+                               device='gpu',
+                               interp_type=types.INTERP_LINEAR,
+                               resize_shorter=resize_uniform)
+    output = fn.crop_mirror_normalize(resized_images,
+                                      device='gpu',
+                                      dtype=types.FLOAT16,
+                                      crop=(224, 224),
+                                      mean=[128., 128., 128.],
+                                      std=[1., 1., 1.],
+                                      mirror=mirror,
+                                      crop_pos_x=uniform[0],
+                                      crop_pos_y=uniform[1])
+    return output
+
+
+def run_benchmark(pipe_fun, batch_size, num_threads, num_samples, debug, data_path):
+    num_iters = num_samples // batch_size
+    times = np.empty(num_iters + 2)
+
+    times[0] = time()
+    pipe = pipe_fun(data_path, batch_size=batch_size, num_threads=num_threads, debug=debug)
+    pipe.build()
+    times[1] = time()
+
+    for i in range(num_iters):
+        pipe.run()
+        times[i + 2] = time()
+
+    full_time = times[-1] - times[1]
+    times = np.diff(times)
+
+    return full_time, times[0 + debug], times[1:]
```

Comment:
So we treat the time of a first run as a build time of the debug pipline, right?

Reply:
done